run-install.sh

```diff
@@ -158,7 +157,19 @@ if [ "$(uname)" = "Darwin" ]; then
         log_message "Homebrew not found. Installing Homebrew..."
         /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
     fi
-    brew install python@3.10
+
+    # Check installed Python version and install the correct Homebrew Python version (macOS)
+    python_version=$(python3 --version | awk '{print $2}' | cut -d'.' -f1,2)
+    if [ "$python_version" = "3.9" ]; then
+        log_message "Python 3.9 detected. Installing Python 3.10 using Homebrew..."
+        brew install python@3.10
+        export PATH="/opt/homebrew/opt/python@3.10/bin:$PATH"
+    elif [ "$python_version" != "3.10" ]; then
+        log_message "Unsupported Python version detected: $python_version. Please use Python 3.10."
+        exit 1
+    fi
+
+    brew install faiss
```

Comment:
why this?

Reply:
<img width="721" alt="Captura de pantalla 2024-12-06 a la(s) 6 45 36 a m" src="https://github.com/user-attachments/assets/73279669-4d64-4413-b136-49004a2d7359">
I thought it would be nice if the user didn't have to download it manually when seeing some messages from the server by doing the brew faiss manually and to make it easier to add it to the code, he won't have to do it manually if it's already predetermined. and with an export PATH="/opt/homebrew/bin:$PATH"

Reply:
<img width="682" alt="Captura de pantalla 2024-12-06 a la(s) 6 47 59 a m" src="https://github.com/user-attachments/assets/f9b089d2-4def-4593-995c-aed94481385f">
<img width="679" alt="Captura de pantalla 2024-12-06 a la(s) 6 48 48 a m" src="https://github.com/user-attachments/assets/7e91f0be-5fcc-4ba0-9cbd-c6f90532e981">